dts/arm/atmel/samx7x.dtsi

```diff
@@ -0,0 +1,482 @@
+/*
+ * Copyright (c) 2024 Henrik Brix Andersen <henrik@brixandersen.dk>
+ * Copyright (c) 2017 Piotr Mienkowski
+ * Copyright (c) 2017 Justin Watson
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <zephyr/dt-bindings/adc/adc.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+#include <zephyr/dt-bindings/clock/atmel_sam_pmc.h>
+
+/ {
+	aliases {
+		watchdog0 = &wdt;
+	};
+
+	chosen {
+		zephyr,flash-controller = &eefc;
+		zephyr,entropy = &trng;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m7";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv7m-mpu";
+				reg = <0xe000ed90 0x40>;
+			};
+		};
+	};
+
+	sram0: memory@20400000 {
+		compatible = "mmio-sram";
+	};
+
+	soc {
+		pmc: pmc@400e0600 {
```

Comment:
rework in ascending order all the peripherals.

Reply:
This is simply a refactoring. I would like to keep it as that. I plan to follow up with a couple of additional PRs, one of which will reorder the nodes in the `samx7x.dtsi` file.

Reply:
Follow-up PR: https://github.com/zephyrproject-rtos/zephyr/pull/83702